### DISCUSSION TITLE
How can I rename HA entity_id when renaming Z2M friendly_name through API?

### DISCUSSION THREAD

**Discussion by SHxKM** [Koenkk/zigbee2mqtt#10672] [General]
I have an existing group:

```
Z2M friendly_name: foo_group
HA entity_id: foo_group
```

When I send this payload to `zigbee2mqtt/bridge/request/group/rename`:

```json
{"from": "foo_group", "to": "bar_group"}
```

Now we have the following:

```
Z2M friendly_name: bar_group
HA entity_id: foo_group
```

Is there anything I can send in the payload to also override the HA `entity_id`?

**Comment by Koenkk** [comment]
I see it is missing from the docs, but I believe this is the same as renaming a device.

https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request

> In case you are using Home Assistant discovery and also want to update the entity ID according to this new name, send e.g. {"from": "my_bulb", "to": "my_bulb_new_name","homeassistant_rename":true}


Could you check if it works?

**Reply by SHxKM** [reply]
> > In case you are using Home Assistant discovery and also want to update the entity ID according to this new name, send e.g. {"from": "my_bulb", "to": "my_bulb_new_name","homeassistant_rename":true}
> 
> Could you check if it works?

That does not work. I tried with or without restarting HA after the MQTT message is sent.

If I restart **both** Z2M and HA then the **friendly_name** in HA changes, but the `entity_id` stays the same.

**Reply by Koenkk** [reply]
It was indeed not supported yet, added this now and updated the docs: https://github.com/Koenkk/zigbee2mqtt.io/commit/a4dd8873a04920461ec1425faaf8e74379c41e99

@nurikk could you also add this to the frontend?

**Reply by SHxKM** [reply]
> It was indeed not supported yet, added this now and updated the docs: [Koenkk/zigbee2mqtt.io@a4dd887](https://github.com/Koenkk/zigbee2mqtt.io/commit/a4dd8873a04920461ec1425faaf8e74379c41e99)
> 
> @nurikk could you also add this to the frontend?

Thanks! Will only be able to test once on the release-branch.

**Comment by sefininio** [comment]
Hi @Koenkk ,

Is there a way to apply this to devices as well?

For example - I have a device with 4 switches. When I changed the device friendly name, it was also renamed in HA, and the related entity ids as well, but it becomes cumbersome.
So a device with a friendly name of "kitchen lights" that has 4 switches:
<img width="680" alt="image" src="https://user-images.githubusercontent.com/7566666/206921441-47e9a3ea-a31a-4bc0-a51e-204b4ff20523.png">

But the HA switches get entity ids like `switch.kitchen_lights_l1`, `switch.kitchen_lights_l2`, etc. It would be better if we could define them as `switch.kitchen_island_lights` for example.

or a kitchen leak sensor device:
<img width="534" alt="image" src="https://user-images.githubusercontent.com/7566666/206921858-f600c7b8-c9b9-430b-884b-bb2432ed0d68.png">

gets an entity_id of `binary_sensor.kitchen_leak_sensor_water_leak`.

Renaming those ids via HA loses sync with mqtt, obviously.
So, is there any way to rename the HA entity ids? 
I saw that in the configuration.yaml there is a `homeassistant` section we can use for each device, but I could not find any information on what can be defined there, so not sure if that is even an option...

**Comment by h0m3d0m0** [comment]
any update on this @Koenkk.
Moved from zwave to zigbee and from deconz to zigbee2mqtt and really enjoy this.
The only thing i missed is the identity id's cannot renamed.
In zwavejs i have the option to rename the id's and this was very helpfully because all my scripting is based on the friendly id name.
Like the example above in have a 4 button light switch.
The device friendly name is "Schakelaar_Living"
But the Id's name's are "Wissel_Eetplaats", "Alles_Uit", "Licht_Living",Licht_MovieCenter"
As you can see i used the switch for many purposes.
But now it is Schakelaar_Living_L1 to L4 so i cannot know by the name whats the meaning is of the button

**Reply by lougreenwood** [reply]
@Koenkk 

If I understand this correctly, this would be super useful for handling SML001 getting the incorrect entity names in HA (e.g `motion` is named `occupancy` and then we have `illuminance` & `illuminance_lux` which I usually re-map to `illuminance_raw` & `illuminance`).
